PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/intellij/runner/functions/component/AppServicePlanPanel.java

```diff
@@ -153,21 +151,14 @@ private void createAppServicePlan() {
         cbAppServicePlan.setSelectedItem(null);
         cbAppServicePlan.setPopupVisible(false);
         final NewAppServicePlanDialog dialog = new NewAppServicePlanDialog(subscriptionId);
-        dialog.pack();
-        dialog.setLocationRelativeTo(this);
-        dialog.addWindowListener(new WindowAdapter() {
-            @Override
-            public void windowClosed(WindowEvent windowEvent) {
-                super.windowClosed(windowEvent);
-                final AppServicePlanWrapper newCreateAppServicePlan = dialog.getAppServicePlan();
-                if (newCreateAppServicePlan != null) {
-                    selectedAppServicePlan = newCreateAppServicePlan;
-                    appServicePlanWrapperList.add(selectedAppServicePlan);
-                }
-                reloadAppServicePlan();
+        if (dialog.showAndGet()) {
+            final AppServicePlanWrapper newCreateAppServicePlan = dialog.getAppServicePlan();
+            if (newCreateAppServicePlan != null) {
+                selectedAppServicePlan = newCreateAppServicePlan;
+                appServicePlanWrapperList.add(selectedAppServicePlan);
             }
-        });
-        dialog.setVisible(true);
+        }
+        reloadAppServicePlan();
```

Comment:
should we move reloadAppServicePlan to inner block of `if (dialog.showAndGet()) `

Reply:
reloadAppServicePlan() will reset the combobox selected item, which works both new app service plan or cancel